learning_observer/learning_observer/auth/social_sso.py

```diff
@@ -210,6 +239,35 @@ async def _process_student_documents(student):
             await _process_student_documents(student)
     # TODO saved skipped doc ids somewhere?
 
+async def _canvas(request):
+    '''
+    Handle Canvas authorization
+    '''
+    if 'error' in request.query:
+        return {}
+    
+    default_server = settings.pmss_settings.default_server(types=['lms', 'canvas_oauth'])
+    
+    url = f'https://{default_server}/login/oauth2/token'
+    common_params = {
+        "grant_type": "refresh_token",
+        'client_id': settings.pmss_settings.client_id(types=['lms', 'canvas_oauth']),
+        'client_secret': settings.pmss_settings.client_secret(types=['lms', 'canvas_oauth']),
+        "refresh_token": settings.pmss_settings.refresh_token(types=['lms', 'canvas_oauth'])
+    }
+    params = common_params.copy()
```

Comment:
Why are we copying the params? This looks like code was copied from the `_google` function without reasoning why it is the way it is.

Reply:
I see that I am not modifying the params object unlike in `_google` function so I removed the copying